lib/generators.nix

```diff
@@ -50,4 +50,18 @@ in
     in
       # map input to ini sections
       mapAttrsToStringsSep "\n" mkSection attrsOfAttrs;
+
+
+  /* Generates JSON from an arbitrary (non-function) value.
+    * For more information see the documentation of the builtin.
+    */
+  toJSON = {}: builtins.toJSON;
+
+
+  /* YAML has been a strict superset of JSON since 1.2, so we
+    * use toJSON. Before it only had a few differences referring
+    * to implicit typing rules, so it should work with older
+    * parsers as well.
+    */
+  toYAML = {}@args: toJSON args;
```

Comment:
No, it doesn’t. Just the JSON subset.

There are generator functions that are surjective (can produce every config value) and some that are “surjective enough”.
We should really annotate which generators are what, since when you create a module you can leave out the `extraConfig` string option if they are surjective, and probably also if they are “surjective enough“, but not if one might need a config that uses features that can’t be expressed with nix expressions.

As an example: YAML files can have internal cross-references. We can’t create them with our toYAML, *but* we can create variables, which effectively creates copies in the final config, but is semantically the same as a cross reference. That’s what I mean with “surjective enough”.